fondant/dataset.py

```diff
@@ -76,60 +105,124 @@ def load_dataframe(self, spec: FondantComponentSpec) -> dd.DataFrame:
 
     def write_subsets(self, df: dd.DataFrame, spec: FondantComponentSpec):
-        for name, subset in spec.output_subsets.items():
-            fields = list(subset.fields.keys())
-            # verify fields are present in the output dataframe
-            subset_columns = [f"{name}_{field}" for field in fields]
+        """
+        Write subset tasks to a remote location
+
+        Args:
+            df (dask.dataframe.DataFrame): The input Dask dataframe.
+            spec (FondantComponentSpec): The specification of the output subsets.
+
+        Raises:
+            ValueError: If a field defined in an output subset is not present in the input
+             dataframe.
+        """
+
+        def verify_subset_columns(subset_name, subset_fields, df):
+            """
+            Verify that all the fields defined in the output subset are present in
+            the input dataframe
+            """
+
+            field_names = list(subset_fields.keys())
+            subset_columns = [f"{subset_name}_{field}" for field in field_names]
             subset_columns.extend(self.mandatory_subset_columns)
 
             for col in subset_columns:
                 if col not in df.columns:
                     raise ValueError(
-                        f"Field {col} defined in output subset {name} but not found in dataframe"
+                        f"Field {col} defined in output subset {subset_name} "
+                        f"but not found in dataframe"
                     )
 
-            # load subset dataframe
+            return subset_columns
+
+        def create_subset_dataframe(subset_name, subset_columns, df):
+            """
+            Create subset dataframe to save with the original field name as the column name
+            """
+            # Create a new dataframe with only the columns needed for the output subset
             subset_df = df[subset_columns]
-            # remove subset prefix from subset columns
+
+            # Remove the subset prefix from the column names
+            prefix_to_replace = f"{subset_name}_"
             subset_df = subset_df.rename(
                 columns={
-                    col: col.split("_")[-1]
+                    col: col[(len(prefix_to_replace)) :]
                     for col in subset_df.columns
                     if col not in self.mandatory_subset_columns
+                    and col.startswith(prefix_to_replace)
                 }
             )
-            # upload to the cloud
-            self._upload_subset(name, subset.fields, subset_df)
+
+            return subset_df
+
+        upload_subsets_tasks = []
+
+        # Loop through each output subset defined in the spec
+        for subset_name, subset in spec.output_subsets.items():
+            # Verify that all the fields defined in the output subset are present in the
+            # input dataframe
+            subset_columns = verify_subset_columns(subset_name, subset.fields, df)
```

Comment:
I think we'll need some logic here for `additionalFields` as well, but it's fine to pick this up in a separate PR. I added a ticket to the project for this.